### ISSUE TITLE
scanr1 and scanl1 on empty vector

### ISSUE THREAD

**Issue by zliu41** [haskell/vector#395]
`scanr1` and `scanl1` currently crash on empty vector. This is different than `List.scanr1/scanl1` which return the empty list. This partiality doesn't seem necessary.

If this is indeed deliberate, it would be useful to give some explanation in Haddock, as well as improve the error message. Currently the message of `scanr1` on empty vector is like

```
*** Exception: ./Data/Vector/Fusion/Stream/Monadic.hs:1298 (scanl1M): empty stream
```

Note it mentions `scanl1M`, rather than `scanr1`, which can be confusing.

**Comment by gksato** [comment]
The fix is in #382, which is already merged, but it hasn't been published yet.

**Comment by zliu41** [comment]
Ah OK, good. Thanks!
